spec/spec_tests/corpus_legacy_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'spec_helper'
+require 'runners/corpus_legacy'
+
+describe 'Driver BSON Corpus Legacy spec tests' do
+
+  BSON_CORPUS_LEGACY_TESTS.each do |path|
+    basename = File.basename(path)
+    # All of the tests in the failures subdir are failing apparently
+    #basename = path.sub(/.*corpus-tests\//, '')
+
+    spec = BSON::CorpusLegacy::Spec.new(path)
+
+    context("(#{basename}): #{spec.description}") do
+
+      spec.valid_tests.each do |test|
+
+        context("VALID CASE: #{test.description}") do
+
+          it 'roundtrips the given bson correctly' do
+            expect(test.reencoded_bson).to eq(test.correct_bson)
+          end
+
+          context 'when the canonical bson is roundtripped', if: test.test_canonical_bson? do
+
+            it 'encodes the canonical bson correctly' do
+              expect(test.reencoded_canonical_bson).to eq(test.correct_bson)
+            end
+          end
+
+          context 'when the document can be represented as extended json', if: test.test_extjson? do
+
+            it 'decodes from the given bson, then encodes the document as extended json correctly' do
+              expect(test.extjson_from_bson).to eq(test.correct_extjson)
+              expect(test.extjson_from_bson[test.test_key]).to eq(test.correct_extjson[test.test_key])
+            end
+
+            it 'decodes from extended json, then encodes the document as extended json correctly' do
+              expect(test.extjson_from_encoded_extjson).to eq(test.correct_extjson)
+              expect(test.extjson_from_encoded_extjson[test.test_key]).to eq(test.correct_extjson[test.test_key])
+            end
+
+            it 'can be converted to extended json using the #as_json method', if: spec.klass.respond_to?(:as_json) do
```

Comment:
It does not appear that any of these blocks actually run. I replaced them with what I had in mind originally.